pkg/source/reconciler/source/kafkasource.go

```diff
@@ -165,6 +172,74 @@
 	src.Status.MarkDeployed(ra)
 	src.Status.CloudEventAttributes = r.createCloudEventAttributes(src)
 
+	logging.FromContext(ctx).Debugf("we have a RA deployment")
+
+	// We need to get all the pods for that ra deployment
+	pods, err := r.KubeClientSet.CoreV1().Pods(src.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: resources.LabelSelector(src.Name),
+	})
```

Comment:
I would use a lister to hit the cache

Reply:
I think It might be useful to include that logic in `control/reconciler`

Reply:
Check out https://github.com/slinkydeveloper/control-data-plane-communication/pull/10